std/algorithm/comparison.d

```diff
@@ -619,6 +619,34 @@ Note:
 auto cmp(R1, R2)(R1 r1, R2 r2)
 if (isInputRange!R1 && isInputRange!R2)
 {
+    version (BigEndian)
+        enum endianForMemcmp = true;
+    else
+        enum endianForMemcmp = ElementEncodingType!R1.sizeof == 1;
+
+    static if (endianForMemcmp
+        && isDynamicArray!R1 && isDynamicArray!R2
+        && is(R1 : E1[], E1) && is(R2 : E2[], E2) // just to get E1 & E2
+        && __traits(isUnsigned, E1) && __traits(isUnsigned, E2)
+        && E1.sizeof == E2.sizeof
+        && (is(Unqual!E1 == char) == is(Unqual!E2 == char)) // Auto-decoding needs to match.
+        && !is(Unqual!E1 == wchar) && !is(Unqual!E2 == wchar) // Can't memcmp UTF-16 to get order.
+        && !is(E1 == __vector) && !is(E2 == __vector))
```

Comment:
Maybe scanner is expecting `!is(E1 == __vector(T1[N1),T1,N1)  && !is(E2 == __vector(T2[N2),T2,N2))`?